### ISSUE TITLE
ERSPAN decapsulation

### ISSUE THREAD

**Issue by mpihelgas** [theclam/stripe#10]
Hi,

Stripe seems to be a really fast tool for decapsulating large PCAPs. 
I have a lot of of large (13GB) PCAPs with GRE and ERSPAN headers, which I would like to decapsulate. Would it be possible to add Cisco ERSPAN decapsulation support to stripe as well? 
I might be mistaken, but I think it currently just strips the GRE header and writes a broken packet which starts with the ERSPAN header.

![image](https://user-images.githubusercontent.com/16393030/116085393-7190db00-a6a7-11eb-9eaa-180c53fb6819.png)
Image taken from: https://infogovworld.com/wp-content/uploads/2019/09/pg18-Layer-3-Routing.jpg

Thank you in advance!

**Comment by theclam** [comment]
Hi.

It looks like it should be fairly straightforward to add this functionality, would you be able to share a pcap file (just a couple of packets would be fine) so I can see if your capture matches the RFC I'm looking at?

Thanks,

Foeh

**Comment by mpihelgas** [comment]
Hi,
Thank you for getting back so quickly! I have attached a test PCAP file (10 packets) below. Zipped it because attaching a .pcap extension was not allowed. Let me know if you need more to make sure it works.

[erspan-test.pcap.zip](https://github.com/theclam/stripe/files/6383160/erspan-test.pcap.zip)

Thank you in advance!

**Comment by theclam** [comment]
Hi,

I've just pushed an updated source code (not done the Windows binary yet) - could you please give that a try?  It seems to work on the small sample PCAP.

Thanks,

Foeh

**Comment by mpihelgas** [comment]
Hi,

I tested with a single 13GB file and it seemed to work nicely. Shaved off about 672MB of headers.
I will try with a larger set of files later.

Thank you again for this quick contribution!
Mauno

**Comment by mpihelgas** [comment]
Hi,
I'm closing this issue. Managed to decapsulate a whole bunch of PCAPs without issues. All the same type of ERSPAN though, so not able to test the other types.
Just a side note, I had to disable/comment the VXLAN decapsulation part because it was decapsulating all Docker Swarm traffic as well, which was something I wanted to remain intact. So just an idea for future features would be to optionally enable/disable some decapsulation functions from cmdline arguments.

But thank you again! Stripe is a great tool -- I'll spread the word.

Best regards,
Mauno
